[PyTorch] Autoformat c10

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56830 [PyTorch] Autoformat c10**
* #57070 [PyTorch] Include hip_runtime.h in macros.h

Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

swolchok committed:
[PyTorch] Autoformat c10

Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

[ghstack-poisoned]

swolchok committed:
forgot to amend formatting changes on "[PyTorch] Autoformat c10"


Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

[ghstack-poisoned]

swolchok committed:
reformat on "[PyTorch] Autoformat c10"


Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

[ghstack-poisoned]

ezyang commented:
did this... break the windows build? Unbelievable.

```
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(91): error C3646: 'ModuleName': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(91): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(92): error C3646: 'hFile': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(92): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(93): error C3646: 'MappedAddress': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(93): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(95): error C3646: 'FileHeader': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(95): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(99): error C3646: 'LastRvaSection': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(99): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(100): error C3646: 'NumberOfSections': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(100): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(101): error C3646: 'Sections': unknown override specifier
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\dbghelp.h(101): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
```

swolchok committed:
suppress clang-format in win32-headers.h on "[PyTorch] Autoformat c10"


Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Autoformat c10"


Opt into formatting on GitHub and format everything. This is a trial run before turning on formatting for more and eventually all of the codebase.

Differential Revision: [D27979080](https://our.internmc.facebook.com/intern/diff/D27979080/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D27979080/)!

[ghstack-poisoned]

mruberry commented:
We've tried to extend autoformatting of C++ several times and always had to revert it. But stepping back, what's the benefit of autoformatting C++? Is it really worth the hit to developer efficiency to comply with the formatter and maintain it? Not to mention the one-time cost to nice blames.
